### ISSUE TITLE
Add DotCover Report

### ISSUE THREAD

**Issue by mholo65** [cake-build/cake#724]
Once DotCover Cover command is implemented #722 , there will be use cases for Merge, Report and Delete. https://www.jetbrains.com/dotcover/help/dotCover__Console_Runner_Commands.html

**Comment by mholo65** [comment]
I was thinking of starting implementing this (or at least Report command). Any objections @patriksvensson?

By adding DotCover Report, I can change my cake-script from:
```
Task("Run-Unit-Tests")
    .IsDependentOn("Build")
    .Does(() =>
{
    if(isRunningOnTeamCity)
    {
        DotCoverCover(tool => {
            tool.NUnit3("/path/to/*.Test.dll",
                new NUnit3Settings {
                    TeamCity = isRunningOnTeamCity
                });
            },
            coverageResult,
            new DotCoverCoverSettings()
                .WithFilter("+:Foo.*")
        );
    }
    else
    {
        DotCoverAnalyse(tool => {
            tool.NUnit3("/path/to/*.Test.dll",
                new NUnit3Settings {
                    TeamCity = isRunningOnTeamCity
                });
            },
            coverageResult,
            new DotCoverAnalyseSettings {
                ReportType = DotCoverReportType.HTML
            }.WithFilter("+:Foo.*")
        );
    }
})
.Finally(() =>
{
    if(isRunningOnTeamCity)
    {
        TeamCity.ImportDotCoverCoverage(coverageResult, MakeAbsolute(Directory("./tools/JetBrains.dotCover.CommandLineTools/tools")));
    }
});
```

To:
```
Task("Run-Unit-Tests")
    .IsDependentOn("Build")
    .Does(() =>
{
    DotCoverCover(tool => {
        tool.NUnit3("/path/to/*.Test.dll",
            new NUnit3Settings {
                TeamCity = isRunningOnTeamCity
            });
        },
        coverageResult,
        new DotCoverCoverSettings()
            .WithFilter("+:Foo.*")
    );
})
.Finally(() =>
{
    if(isRunningOnTeamCity)
    {
        TeamCity.ImportDotCoverCoverage(coverageResult, MakeAbsolute(Directory("./tools/JetBrains.dotCover.CommandLineTools/tools")));
    }
    else
    {
        DotCoverReport(coverageResult, "/path/to/output.html");
    }
});
```

And eliminate risk of running with different DotCoverSettings on TeamCity vs. Local

**Comment by gep13** [comment]
@mholo65 you mentioned in your last PR that this partially solves this issue.  Is there still work that needs to be done in order to close out this issue?

**Comment by mholo65** [comment]
@gep13 well, implement Delete and Merge... 😄 

1. Currently, I don't see use case for Delete at least... 
1. Merge might be useful in complex scenarios (if having multiple snapshot files that needs to be merged prior to running **TeamCity import** or **DotCover Report**).

**Comment by mholo65** [comment]
@gep13 as I see it, we could rename this issue and create separate issue for Merge..? Thoughts?

**Comment by gep13** [comment]
@mholo65 sounds good to me.

I am just signing off for the night.  Can I ask that you do that, and then close this issue when you are happy?

Thanks!

**Comment by mholo65** [comment]
Implemented in #1153
